spec/services/create_next_edition_service_spec.rb

```diff
@@ -0,0 +1,118 @@
+RSpec.describe CreateNextEditionService do
+  describe ".call" do
+    let(:user) { create :user }
+
+    before do
+      populate_default_government_bulk_data
+    end
+
+    it "aborts if the current edition isn't live" do
+      current_edition = create(:edition, live: false)
+
+      expect { described_class.call(current_edition: current_edition, user: user) }
+        .to raise_error("Can only create a next edition from a live edition")
+    end
+
+    it "updates the documents current edition" do
+      current_edition = create(:edition, :published)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.current).to be(true)
+    end
+
+    it "returns a draft edition" do
+      current_edition = create(:edition, :published)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.draft?).to be(true)
+    end
+
+    it "resets change_note, update_type, and propoposed_publish_time values" do
+      current_edition = create(:edition, :published)
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_note).to eq("")
+      expect(next_edition.update_type).to eq("major")
+      expect(next_edition.proposed_publish_time).to eq(nil)
+    end
+
+    it "appends the change note details when there is a change note and a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "major")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history.first).to match a_hash_including(
+        "note" => "note",
+        "public_timestamp" => Date.yesterday.noon.rfc3339,
+      )
+    end
+
+    it "maintains the change history when the current edition is not a major change" do
```

Comment:
Should we mention in the `it` how the edition contains a change note too as it's a little surprising from reading the it statement that the edition also contains a change note when it's a minor change? The example above mentions both `it "appends the change note details when there is a change note and a major change" do`

Reply:
@1pretz1 Thanks, yes - this is a little clunky.  Have updated this to `it "does not update the change history when the current edition is not a major change" do` and the following test (with a similar issue) to `it "does not update the change history when the current edition lacks a change note" do`.